articles/openshift/oshift-ref-security-best-practices.md

```diff
@@ -0,0 +1,112 @@
+
+You should ensure that OpenShift account credentials are not shared between multiple users. Instead, you should create individual user accounts for each user with the correct role binding. It is also the responsibility of each individual user to ensure the physical security of the terminal from which they access OpenShift.
+
+## Service accounts
+
+You should use service accounts for all API access, which should be accomplished by an automated process; never use a regular user's credentials. Usually a service account's username is the same as the project it is created for, followed by its name:
+
+```bash
+system:serviceaccount:<project>:<name>. 
+```
+
+For example:
+
+```bash
+system:serviceaccount:continuous-integration:jenkins
+```
+
+You can read more about service accounts [here](https://docs.openshift.com/container-platform/3.11/dev_guide/service_accounts.html).
+
+## Creating and deploying vulnerable container images
+
+There have been many incidents in recent history whereby containers that contain severe vulnerabilities have been deployed into production clusters. To avoid this from happening, you should use base images from trusted sources, that is, from Red Hat's container registry. These containers are scanned for vulnerabilities on a daily basis and given a security rating based on known security risks at that moment in time. Docker Hub also provides some certified containers. Users can also choose to build their own base containers from scratch. However, containers built from scratch could still contain vulnerable packages. Therefore, it is good practice to set up an automated image scanning process, as part of your container building and deployment process, which can scan the built image and provide an image security rating. It is also possible to use image scanning tools to certify your own images before they are deployed so that only images with an acceptable security rating set by your OpenShift administrators are deployed.
+
+Some container image scanning tools include:
+
+  -  OpenSCAP-docker
+
+  -  Black Duck Hub
+
+  -  JFrog Xray
+
+  -  Twistlock
+
+  -  Image-inspector
+
+  -  Clair
+
+You should also apply the Principle of least privilege with containers too. If it can be avoided you should never run containers as the root user; instead always run as a user with just enough privileges to run the required container workload.
+
+## Secrets
+
+Sensitive information, such as passwords, SSH keys, tokens and certificates, are examples of information that could be considered secret. Such secrets should never be stored in any place in readable format (for example, in source code, GitHub repositories, and so on). It is imperative that when secrets are required in OpenShift, they are set up as OpenShift Secrets. This allows the secrets to be encrypted within OpenShift and accessed only by the relevant containers when they are needed.
```

Comment:
I think we should expand this section and point out the limitations of secrets (e.g. base64 encoded, potentially accessible from any container on the same node). Whilst its a tool to get started with there are better secret options out there like vault that we could be recommending.